janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/hbase/HBaseSnapshotBinaryInputFormat.java

```diff
@@ -0,0 +1,171 @@
+
+    @Override
+    public void setConf(final Configuration config) {
+        super.setConf(config);
+
+        // Pass the extra pass-through properties directly to HBase/Hadoop config.
+        Map<String, Object> configSub =
+                janusgraphConf.getSubset(HBaseStoreManager.HBASE_CONFIGURATION_NAMESPACE);
+        for (Map.Entry<String, Object> entry : configSub.entrySet()) {
+            log.info("HBase configuration: setting {}={}", entry.getKey(), entry.getValue());
+            if (entry.getValue() == null) continue;
+            config.set(entry.getKey(), entry.getValue().toString());
+        }
+
+        config.set("autotype", "none");
+        Scan scanner = new Scan();
+        String cfName = mrConf.get(JanusGraphHadoopConfiguration.COLUMN_FAMILY_NAME);
+        // TODO the space-saving short name mapping leaks from HBaseStoreManager here
+        if (janusgraphConf.get(HBaseStoreManager.SHORT_CF_NAMES)) {
+            try {
+                final BiMap<String, String> shortCfMap =
+                        HBaseStoreManager.createShortCfMap(janusgraphConf);
+                cfName = HBaseStoreManager.shortenCfName(shortCfMap, cfName);
+            } catch (PermanentBackendException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        edgeStoreFamily = Bytes.toBytes(cfName);
+        scanner.addFamily(edgeStoreFamily);
+
+        // This is a workaround, to be removed when convertScanToString becomes public in hbase
+        // package.
+        Method converter;
+        try {
+            converter =
+                    TableMapReduceUtil.class.getDeclaredMethod("convertScanToString", Scan.class);
+            converter.setAccessible(true);
+            config.set(TableInputFormat.SCAN, (String) converter.invoke(null, scanner));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        String snapshotName = janusgraphConf.get(HBaseStoreManager.HBASE_SNAPSHOT);
+        String restoreDirString = janusgraphConf.get(HBaseStoreManager.HBASE_SNAPSHOT_RESTORE_DIR);
+        Path restoreDir = new Path(restoreDirString);
+        try {
+            // This is a workaround. TableSnapshotInputFormat.setInput accepts a Job as parameter.
+            // And the Job.getInstance(config) create clone of the config, not setting on the
+            // passed in config.
+            Job job = Job.getInstance(config);
+            TableSnapshotInputFormat.setInput(job, snapshotName, restoreDir);
+            config.set(SNAPSHOT_NAME_KEY, job.getConfiguration().get(SNAPSHOT_NAME_KEY));
+            config.set(RESTORE_DIR_KEY, job.getConfiguration().get(RESTORE_DIR_KEY));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public RecordReader<ImmutableBytesWritable, Result> getTableReader() {
+        return tableReader;
+    }
+
+    public byte[] getEdgeStoreFamily() {
+        return edgeStoreFamily;
+    }
```

Comment:
`getEdgeStoreFamily` and `getTableReader` don't appear to be used, can they be removed?

Reply:
Ok

Reply:
Ok